services/venice-server/src/main/java/com/linkedin/venice/listener/ReadQuotaEnforcementHandler.java

```diff
@@ -300,12 +315,13 @@ public boolean handleTooManyRequests(
     return true;
   }
 
-  public void handleEnforcingAndNoBucket(RouterRequest request) {
-    LOGGER.warn(
-        "Request for resource: {} but no TokenBucket for that resource. Not yet enforcing quota",
-        request.getResourceName());
+  /**
+   * This method and the expiring set noBucketStores is only used to throttle the logging of such event
+   */
+  public void handleEnforcingAndNoBucket(String resourceName) {
+    LOGGER.warn("Request for resource: {} but no TokenBucket for that resource. Not yet enforcing quota", resourceName);
```

Comment:
could we add a rate limiter log for this? during the edge case there could be many such cases.

Reply:
Good call, will do.

Reply:
Actually this is already rate limited to 30 seconds per resource with the `noBucketStores` expiring set.